lib/builder/util.go

```diff
@@ -63,15 +65,24 @@ func checkBuildEnv() error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	_, err = client.Version()
-	if err != nil {
-		log.Error(trace.DebugReport(err))
-		return trace.BadParameter(
-			`Docker does not seem to be available on this machine.
+	dockerErr := trace.BadParameter(
+		`Docker does not seem to be available on this machine.
 To resolve the issue:
   * Install Docker (https://docs.docker.com/engine/installation/).
   * Make sure it can be used by a non-root user (https://docs.docker.com/install/linux/linux-postinstall/).`)
+
+	_, err = client.Version()
+	if err != nil {
+		log.WithError(err).Error("Failed to validate docker client connectivity.")
+		return dockerErr
 	}
+
+	err = utils.Exec(exec.Command("docker", "version"), ioutil.Discard)
```

Comment:
Fixed in 01533bc.

Here is the new debug output when the binary is missing:

```
ERRO [BUILDER]   failed to validate docker binary, output: "" error:[
ERROR REPORT:
Original Error: *exec.Error exec: &#34;docker&#34;: executable file not found in $PATH         
Stack Trace:
        /home/walt/git/gravity/lib/utils/exec.go:188 github.com/gravitational/gravity/lib/utils.ExecWithInput                                                                         
        /home/walt/git/gravity/lib/utils/exec.go:182 github.com/gravitational/gravity/lib/utils.Exec
        /home/walt/git/gravity/lib/builder/util.go:81 github.com/gravitational/gravity/lib/builder.checkBuildEnv
        /home/walt/git/gravity/lib/builder/engine.go:120 github.com/gravitational/gravity/lib/builder.newEngine
        /home/walt/git/gravity/lib/builder/cluster.go:34 github.com/gravitational/gravity/lib/builder.NewClusterBuilder
        /home/walt/git/gravity/e/tool/tele/cli/build.go:65 github.com/gravitational/gravity/e/tool/tele/cli.buildClusterImage
        /home/walt/git/gravity/e/tool/tele/cli/run.go:52 github.com/gravitational/gravity/e/tool/tele/cli.Run
        /home/walt/git/gravity/e/tool/tele/main.go:31 main.run
        /home/walt/git/gravity/e/tool/tele/main.go:22 main.main
        /usr/local/go/src/runtime/proc.go:204 runtime.main
        /usr/local/go/src/runtime/asm_amd64.s:1374 runtime.goexit
User Message: exec: &#34;docker&#34;: executable file not found in $PATH] builder/util.go:83
ERRO             Command failed. error:[
ERROR REPORT:
Original Error: *trace.BadParameterError Docker does not seem to be available on this machine.
To resolve the issue:
  * Install Docker (https://docs.docker.com/engine/installation/).
  * Make sure it can be used by a non-root user (https://docs.docker.com/install/linux/linux-postinstall/).
Stack Trace:
        /home/walt/git/gravity/lib/builder/util.go:68 github.com/gravitational/gravity/lib/builder.checkBuildEnv
        /home/walt/git/gravity/lib/builder/engine.go:120 github.com/gravitational/gravity/lib/builder.newEngine
        /home/walt/git/gravity/lib/builder/cluster.go:34 github.com/gravitational/gravity/lib/builder.NewClusterBuilder
        /home/walt/git/gravity/e/tool/tele/cli/build.go:65 github.com/gravitational/gravity/e/tool/tele/cli.buildClusterImage
        /home/walt/git/gravity/e/tool/tele/cli/run.go:52 github.com/gravitational/gravity/e/tool/tele/cli.Run
        /home/walt/git/gravity/e/tool/tele/main.go:31 main.run
        /home/walt/git/gravity/e/tool/tele/main.go:22 main.main
        /usr/local/go/src/runtime/proc.go:204 runtime.main
        /usr/local/go/src/runtime/asm_amd64.s:1374 runtime.goexit
User Message: Docker does not seem to be available on this machine.
To resolve the issue:
  * Install Docker (https://docs.docker.com/engine/installation/).
  * Make sure it can be used by a non-root user (https://docs.docker.com/install/linux/linux-postinstall/).] tele/main.go:23
[ERROR]: Docker does not seem to be available on this machine.
To resolve the issue:
  * Install Docker (https://docs.docker.com/engine/installation/).
  * Make sure it can be used by a non-root user (https://docs.docker.com/install/linux/linux-postinstall/).
```